File: unicardsoftwaredevelopment-androidmobileappfortopupandrefund-c0e3674d02c4/app/src/main/java/au/com/unicard/tafenswwallet/ui/topup/TopupFragment.java
package au.com.unicard.tafenswwallet.ui.topup;

import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import au.com.unicard.tafenswwallet.R;

public class TopupFragment extends Fragment {

    private TopupViewModel topupViewModel;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {

        topupViewModel = ViewModelProviders.of(this).get(TopupViewModel.class);
        View root = inflater.inflate(R.layout.fragment_topup, container, false);
        final TextView textView = root.findViewById(R.id.text_topup);

        topupViewModel.getText().observe(this, new Observer<String>() {
            @Override
            public void onChanged(@Nullable String s) {
                textView.setText(s);
            }
        });
        return root;
    }
}

File: TAFEWallet/app/src/main/java/com/yuan/tafewallet/ui/topup/TopupFragment.kt
package com.yuan.tafewallet.ui.topup

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.yuan.tafewallet.MainActivity
import com.yuan.tafewallet.R
import com.yuan.tafewallet.models.Account

class TopupFragment : Fragment(), TopupSelectAccountTableViewAdapter.TopupSelectAccountTableViewClickListener {

    private lateinit var topupViewModel: TopupViewModel
    lateinit var tableView: RecyclerView

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val root = inflater.inflate(R.layout.fragment_topup, container, false)

        return root
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        view?.let {
            tableView = it.findViewById(R.id.TopUpAccountTable)
            tableView.layoutManager = LinearLayoutManager(activity)
            tableView.adapter = TopupSelectAccountTableViewAdapter(this)

        }
    }

    override fun listItemClicked(account: Account) {
        val fragment = TopupSelectAmountFragment.newInstance(account)
        (activity as MainActivity).gotoFragment(fragment, TopupSelectAmountFragment.TAG)
    }

    companion object {
        val TAG = TopupFragment::class.java.simpleName
        @JvmStatic
        fun newInstance() = TopupFragment()
    }
}